### PR TITLE
Provide documentation for some state machine related functions

### PR DESCRIPTION
**Pull request by mattcaswell**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/SSL_in_init.pod [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+ OSSL_HANDSHAKE_STATE SSL_get_state(const SSL *ssl);
+
+=head1 DESCRIPTION
+
+SSL_in_init() returns 1 if the SSL/TLS state machine is currently processing or
+awaiting handshake messages, or 0 otherwise.
+
+SSL_in_before() returns 1 if no SSL/TLS handshake has yet been initiated, or 0
+otherwise.
```
Comment (levitte) [inline doc/man3/SSL_in_init.pod]: A gentle reminder, the comment in that function says this:

``` C
    /*
     * Historically being "in before" meant before anything had happened. In the
     * current code though we remain in the "before" state for a while after we
     * have started the handshake process (e.g. as a server waiting for the
     * first message to arrive). There "in before" is taken to mean "in before"
     * and not started any handshake process yet.
     */
```

I'm fine with the your explanation, as long as we're fine with this small bend of the truth.  Your call.

Comment (mattcaswell) [inline doc/man3/SSL_in_init.pod]: Well, I don't really see the "bend of the truth". Basically there are two different things - the "in before" state in the state machine, and what "SSL_in_before()" returns. The comment is just (trying) to explain the difference between the two things. The function SSL_in_before() returns 1 when no SSL/TLS handshake has yet been initiated (as the documentation states). However the state machine may still be in the "in before" state for a short while after SSL_in_before() no longer returns 1, i.e. the application has asked OpenSSL to start the handshake process, but as yet no messages have actually been processed. In other words I think the docs I have written are consistent with the comment.

**Comment by mattcaswell** [comment]
So this PR duplicates the documentation in #3048. Do we prefer this version?

File: doc/man3/SSL_in_init.pod (hunk 1 above)
Comment (levitte) [inline doc/man3/SSL_in_init.pod]: I'm satisfied with that clarification, thank you.

**Comment by levitte** [comment]
I prefer this to #3048, as this is more fleshed out.  Sorry, @jon-oracle.

**Comment by mattcaswell** [comment]
Pushed. Thanks!

**Pull request closed without merging**